### PR TITLE
Add support for range functions in TileDB-Go

### PR DESCRIPTION
**Pull request by antalakas**
[ch938]

### CODE REVIEW

File: examples/range_test.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+	if _, err := os.Stat(rangeArrayName); err == nil {
+		err = os.RemoveAll(rangeArrayName)
+		checkError(err)
+	}
+
+	// Output: Error adding query range: [TileDB::Subarray] Error: Cannot add range to dimension; Range must be in the domain the subarray is constructed from
+	// Error adding query range: [TileDB::Subarray] Error: Cannot add range to dimension; Invalid dimension index
+	// Number of ranges across dimension 0 is: 1
+	// Range start for dimension 0, range 0 is: 1
+	// Range end for dimension 0, range 0 is: 4
```
Comment (antalakas) [inline examples/range_test.go]: I expected this to be 3

Comment (Shelnutt2) [reply examples/range_test.go]: Ranges are inclusive, and you set `var end int32 = 3`, so I believe 3 is correct.

File: examples/range_test.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
+/**
+ * @file   quickstart_dense_test.go
```
Comment (Shelnutt2) [inline examples/range_test.go]: File name needs to be updated

File: examples/range_test.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2018 TileDB, Inc.
```
Comment (Shelnutt2) [inline examples/range_test.go]: copyright should be 2019

**Comment by Shelnutt2** [comment]
Minor changes but looks good! Feel free to merge after the comment fixes :+1:

**Pull request merged**